### PR TITLE
Changelog entry for OCaml 5.1.0-beta1

### PR DESCRIPTION
**Pull request by Octachron**
This PR adds the changelog entry for the first beta release of OCaml 5.1.0.

### CODE REVIEW

File: data/changelog/ocaml/2023-07-04-ocaml-5.1.beta1.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  ### Windows Bugfix
+
+  - [#12184](https://github.com/ocaml/ocaml/issues/12184), [#12320](https://github.com/ocaml/ocaml/issues/12320): Sys.rename Windows fixes on directory corner cases.
+    (Jan Midtgaard, review by Anil Madhavapeddy)
+
+---
+
+After two alpha releases, the release of OCaml 5.1.0 is drawing near.
+We have thus released a first beta version of OCaml 5.1.0 to help you update your softwares and libraries ahead of the release (see below for the installation instructions).
+Compared to the last alpha release, this beta contains two subtle internal runtime fixes, and one Windows fix.
```
Comment (christinerose) [inline data/changelog/ocaml/2023-07-04-ocaml-5.1.beta1.md]:
```suggestion
Compared to the last alpha release, this beta contains two subtle internal runtime fixes and one Windows fix.
```

File: data/changelog/ocaml/2023-07-04-ocaml-5.1.beta1.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+Overall, the opam ecosystem looks in a good shape for the first beta release.
+
+If you find any bugs, please report them on [OCaml's issue tracker](https://github.com/ocaml/ocaml/issues).
+
+Nearly all core development tools supports OCaml 5.1.0, and you can follow the last remaining wrinkles on the
```
Comment (christinerose) [inline data/changelog/ocaml/2023-07-04-ocaml-5.1.beta1.md]:
```suggestion
Nearly all core development tools support OCaml 5.1.0, and you can follow the last remaining wrinkles on the
```

**Review by christinerose — APPROVED** [review]
Just two minor grammar fixes. Looks good!

**Comment by tmattio** [comment]
Thank you @Octachron, and sorry for the delay to merge, this slipped through.

**Pull request merged**